miner/worker.go

```diff
@@ -225,6 +227,7 @@ func newWorker(config *Config, chainConfig *params.ChainConfig, engine consensus
 		recommit = minRecommitInterval
 	}
 
+	worker.wg.Add(4)
 	go worker.mainLoop()
 	go worker.newWorkLoop(recommit)
 	go worker.resultLoop()
```

Comment:
I'm suggesting an approach which I think makes managing the wg a bit easier, by keeping all the code contained in one place.

```suggestion
	worker.wg.Add(4)
	go func() {
		defer worker.wg.Done()
		worker.mainLoop()
	}()
	go func() {
		defer worker.wg.Done()
		worker.newWorkLoop(recommit)
	}()
	go func() {
		defer worker.wg.Done()
		worker.resultLoop()
	}()
	go func() {
		defer worker.wg.Done()
		worker.taskLoop()
	}()
```